src/Experimental/aggregation.jl

```diff
@@ -0,0 +1,197 @@
+using LightGraphs:AbstractPathState, DijkstraState, BellmanFordState, FloydWarshallState, DEsopoPapeState
+import Base:convert, getproperty
+import LightGraphs: enumerate_paths
+
+struct LGEnvironment
+    threaded::Bool
+    parallel::Bool
+    LGEnvironment() = new(false, false)
+end
+     
+abstract type AbstractGraphResults end
+abstract type ShortestPathResults <: AbstractGraphResults end
+abstract type AbstractGraphAlgorithm end
+abstract type ShortestPathAlgorithm <: AbstractGraphAlgorithm end
+
+##################
+##   Dijkstra   ##
+##################
+struct DijkstraShortestPathResults{T<:Real, U<:Integer}  <: ShortestPathResults
+    parents::Vector{U}
+    dists::Vector{T}
+    predecessors::Vector{Vector{U}}
+    pathcounts::Vector{UInt64}
+    closest_vertices::Vector{U}
+end
+
+convert(::Type{AbstractPathState}, spr::DijkstraShortestPathResults) = convert(DijkstraState, spr)
+convert(::Type{<:DijkstraShortestPathResults}, s::DijkstraState) =
+    DijkstraShortestPathResults(s.parents, s.dists, s.predecessors, s.pathcounts, s.closest_vertices)
+
+convert(::Type{<:DijkstraState}, spr::DijkstraShortestPathResults) =
+    DijkstraState(spr.parents, spr.dists, spr.predecessors, spr.pathcounts, spr.closest_vertices)
+
+# These constructors do not copy. They probably should, but you shouldn't be changing the values in any case.
+# Can we just document that it's undefined behavior to change any element of a SPR?
+DijkstraShortestPathResults(s::DijkstraState) = convert(DijkstraShortestPathResults, s)
+DijkstraState(spr::DijkstraShortestPathResults) = convert(DijkstraState, spr)
+
+struct DijkstraShortestPathAlgorithm <: ShortestPathAlgorithm
+    all_paths::Bool
+    track_vertices::Bool
+
+    DijkstraShortestPathAlgorithm() = new(false, false)
+end
+
+
+##################
+##    A-Star    ##
+##################
+struct AStarShortestPathResults{E<:AbstractEdge} <: ShortestPathResults
+    path::Vector{E}
+end
+# for completeness and consistency.
+struct AStarState{E<:AbstractEdge} <: AbstractPathState
+    path::Vector{E}
+end
+
+function enumerate_paths(s::AStarState)
```

Comment:
What about calling this `paths(s::AStarState)`so that the API is 
```julia
s = shortest_paths(g, AStarShortestPathAlgorithm())
for path in paths(s)
    #do something with path
end
```

Reply:
A* is unique in that it returns a single path from `s` to `t`. (It's the only SP algorithm that takes `t`.)

Otherwise, yes - I prefer `paths` to `enumerate_paths` across the board.